### PR TITLE
Prevent SQL injection

### PR DESCRIPTION
**Pull request by crgimenes**
Prevents SQL injection as mentioned in issue #36 (SQL injection is possible) as well other cases where failure could occur.

To correct the problem:
- The SQL calls are now made using prepare statement.
- The paging parameters only accept numbers.
- Where possible the identifiers are validated.

### CODE REVIEW

File: controllers/databases.go [hunk 1]
```diff
@@ -11,7 +11,12 @@ import (
 
 // GetDatabases list all (or filter) databases
 func GetDatabases(w http.ResponseWriter, r *http.Request) {
-	requestWhere := postgres.WhereByRequest(r)
+	requestWhere, values, err := postgres.WhereByRequest(r, 1)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
```
Comment (felipeweb) [inline controllers/databases.go]: missing return

File: controllers/schemas.go [hunk 2]
```diff
@@ -11,7 +11,12 @@ import (
 
 // GetSchemas list all (or filter) schemas
 func GetSchemas(w http.ResponseWriter, r *http.Request) {
-	requestWhere := postgres.WhereByRequest(r)
+	requestWhere, values, err := postgres.WhereByRequest(r, 1)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
```
Comment (felipeweb) [inline controllers/schemas.go]: missing return

File: controllers/tables.go [hunk 3]
```diff
@@ -15,7 +15,12 @@ import (
 
 // GetTables list all (or filter) tables
 func GetTables(w http.ResponseWriter, r *http.Request) {
-	requestWhere := postgres.WhereByRequest(r)
+	requestWhere, values, err := postgres.WhereByRequest(r, 1)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
```
Comment (felipeweb) [inline controllers/tables.go]: missing return

File: controllers/tables.go [hunk 4]
```diff
@@ -51,7 +56,12 @@ func GetTablesByDatabaseAndSchema(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "Unable to parse schema in URI", http.StatusInternalServerError)
 		return
 	}
-	requestWhere := postgres.WhereByRequest(r)
+	requestWhere, values, err := postgres.WhereByRequest(r, 3)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
```
Comment (felipeweb) [inline controllers/tables.go]: missing return

File: controllers/tables.go [hunk 5]
```diff
@@ -96,17 +118,28 @@ func SelectFromTables(w http.ResponseWriter, r *http.Request) {
 	}
 
 	query := fmt.Sprintf("%s %s.%s.%s", statements.SelectInTable, database, schema, table)
-	requestWhere := postgres.WhereByRequest(r)
+	requestWhere, values, err := postgres.WhereByRequest(r, 1)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
```
Comment (felipeweb) [inline controllers/tables.go]: missing return

File: controllers/tables.go [hunk 6]
```diff
@@ -176,9 +209,13 @@ func DeleteFromTable(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	where := postgres.WhereByRequest(r)
+	where, values, err := postgres.WhereByRequest(r, 1)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
```
Comment (felipeweb) [inline controllers/tables.go]: missing return

File: controllers/tables.go [hunk 7]
```diff
@@ -219,7 +255,13 @@ func UpdateTable(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	object, err := postgres.Update(database, schema, table, where, req)
+	where, values, err := postgres.WhereByRequest(r, 1)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
```
Comment (felipeweb) [inline controllers/tables.go]: missing return

**Comment by avelino** [comment]
thank you @crgimenes

**Pull request merged**